crates/rpc/rpc/src/eth/api/changes.rs

```diff
@@ -0,0 +1,76 @@
+use crate::eth::error::{EthApiError, EthResult};
+use crate::EthApi;
+use reth_primitives::{Account, Address, BlockId, U256};
+use reth_provider::{AccountExtReader, BlockReaderIdExt, StateProviderFactory};
+use revm_primitives::HashMap;
+
+impl<Provider, Pool, Network> EthApi<Provider, Pool, Network>
+where
+    Provider: AccountExtReader + BlockReaderIdExt + StateProviderFactory + 'static,
+{
+    pub(crate) fn account_changes_in_block(
+        &self,
+        block_id: impl Into<BlockId>,
+    ) -> EthResult<Option<HashMap<Address, Account>>> {
+        let block_id = block_id.into();
+        let Some(block_number) = self.provider().block_number_for_id(block_id)? else {
+            return Err(EthApiError::UnknownBlockNumber)
+        };
+
+        let changed_accounts =
+            self.provider().changed_accounts_with_range(block_number..=block_number)?;
+        let state = self.provider().state_by_block_id(block_id)?;
+
+        if changed_accounts.is_empty() {
+            return Ok(None);
+        }
+
+        return changed_accounts.iter().try_fold(Some(HashMap::new()), |hash_map, &address| {
+            let Some(account) = state.basic_account(address)? else {
+                return Err(EthApiError::InternalEthError)
+            };
+
+            let mut hash_map = hash_map.unwrap();
+            hash_map.insert(address, account);
```

Comment:
Q, should we include the previous account state in response so that we can have `old` `new` account?

Reply:
Check this: https://github.com/paradigmxyz/reth/pull/3768/files#r1263497541
It explains it how to do it.